bpf/kmesh/workload/include/service.h

```diff
@@ -37,6 +44,53 @@ static inline int lb_random_handle(struct kmesh_context *kmesh_ctx, __u32 servic
     return 0;
 }
 
+static inline int
+lb_locality_failover_handle(struct kmesh_context *kmesh_ctx, __u32 service_id, service_value *service_v, bool is_strict)
+{
+    int ret = 0;
+    uint32_t rand_k = 0;
+    endpoint_key endpoint_k = {0};
+    endpoint_value *endpoint_v = NULL;
+    endpoint_k.service_id = service_id;
+    struct ip_addr zero_addr = {0};
+    __u32 zero_port = 0;
+
+    // #pragma unroll
+    for (int match_prio = 0; match_prio < PRIO_COUNT; match_prio++) {
+        endpoint_k.prio = match_prio; // 0->6
+        // if we have endpoints in this prio
+        if (service_v->prio_endpoint_count[match_prio] > 0) {
+            rand_k = bpf_get_prandom_u32() % service_v->prio_endpoint_count[match_prio] + 1;
+            if (rand_k >= MAP_SIZE_OF_BACKEND) {
```

Comment:
how could this happen

Reply:
If left unrestricted, the BPF program may exhibit anomalies (the exact cause of errors has not been observed yet, but adding this restriction is necessary).